src/Angor/Avalonia/Angor.Contexts.Funding/Shared/PublishTransaction.cs

```diff
@@ -25,9 +28,104 @@ public async Task<Result<string>> Handle(PublishTransactionRequest request, Canc
 
             var errorMessage = await indexerService.PublishTransactionAsync(request.TransactionDraft.SignedTxHex);
             
-            return !string.IsNullOrEmpty(errorMessage) 
-                ? Result.Failure<string>(errorMessage) 
-                : Result.Success(request.TransactionDraft.TransactionId);
+            if (!string.IsNullOrEmpty(errorMessage))
+            {
+                return Result.Failure<string>(errorMessage);
+            }
+
+            // If this is an investment draft and we have wallet/project context, persist to portfolio
+            if (request.TransactionDraft is InvestmentDraft investmentDraft && 
+                request.WalletId.HasValue && 
+                request.ProjectId != null)
+            {
+                var investmentRecord = new InvestmentRecord
+                {
+                    InvestmentTransactionHash = request.TransactionDraft.TransactionId,
+                    InvestmentTransactionHex = request.TransactionDraft.SignedTxHex,
+                    InvestorPubKey = investmentDraft.InvestorKey,
+                    ProjectIdentifier = request.ProjectId.Value,
+                    UnfundedReleaseAddress = null, // No penalty path for direct investments
+                    RequestEventId = null, // No founder approval request for investments below threshold
+                    RequestEventTime = null
+                };
+
+                var addResult = await portfolioService.Add(request.WalletId.Value, investmentRecord);
```

Comment:
At this point we have the record because we added it when sending the offer to the founder, so we know in the portfolio that there is an investment request